src/OfficeService.Validation/WorkspaceType/EditWorkspaceTypeRequestValidator.cs

```diff
@@ -0,0 +1,158 @@
+{
+  private readonly IWorkspaceTypeRepository _workspaceTypeRepository;
+
+  private async Task HandleInternalPropertyValidationAsync(Operation<EditWorkspaceTypeRequest> requestedOperation, CustomContext context)
+  {
+    Context = context;
+    RequestedOperation = requestedOperation;
+
+    Thread.CurrentThread.CurrentUICulture = new CultureInfo("ru-RU");
+
+    #region paths
+
+    AddСorrectPaths(
+      new List<string>
+      {
+        nameof(EditWorkspaceTypeRequest.Name),
+        nameof(EditWorkspaceTypeRequest.Description),
+        nameof(EditWorkspaceTypeRequest.StartTime),
+        nameof(EditWorkspaceTypeRequest.EndTime),
+        nameof(EditWorkspaceTypeRequest.BookingRule),
+        nameof(EditWorkspaceTypeRequest.IsActive)
+      });
+
+    AddСorrectOperations(nameof(EditWorkspaceTypeRequest.Name), new List<OperationType> { OperationType.Replace });
+    AddСorrectOperations(nameof(EditWorkspaceTypeRequest.Description), new List<OperationType> { OperationType.Replace });
+    AddСorrectOperations(nameof(EditWorkspaceTypeRequest.StartTime), new List<OperationType> { OperationType.Replace });
+    AddСorrectOperations(nameof(EditWorkspaceTypeRequest.EndTime), new List<OperationType> { OperationType.Replace });
+    AddСorrectOperations(nameof(EditWorkspaceTypeRequest.BookingRule), new List<OperationType> { OperationType.Replace });
+    AddСorrectOperations(nameof(EditWorkspaceTypeRequest.IsActive), new List<OperationType> { OperationType.Replace });
+
+    #endregion
+
+    #region Name
+
+    await AddFailureForPropertyIfAsync(
+      nameof(EditWorkspaceTypeRequest.Name),
+      x => x == OperationType.Replace,
+      new()
+      {
+        {
+          x => Task.FromResult(!string.IsNullOrEmpty(x.value?.ToString().Trim())),
+          "Workspace type name cannot be empty."
+        },
+        {
+          x => Task.FromResult(x.value.ToString().Trim().Length < 101),
+          "Workspace type name length cannot be more than 100 characters."
+        },
+        {
+          async x => !await _workspaceTypeRepository.DoesNameExistAsync(x.value?.ToString().Trim()),
+          "Workspace type name already exists."
+        }
+      }, CascadeMode.Stop);
+
+    #endregion
+
+    #region Description
+
+    AddFailureForPropertyIf(
+      nameof(EditWorkspaceTypeRequest.Description),
+      x => x == OperationType.Replace,
+      new()
+      {
+        {
+          x => string.IsNullOrEmpty(x.value?.ToString()) || x.value.ToString().Trim().Length < 301,
+          "Workspace type description length cannot be more than 300 characters."
+        }
+      });
+
+    #endregion
+
+    #region StartTime
+
+    AddFailureForPropertyIf(
+      nameof(EditWorkspaceTypeRequest.StartTime),
+      x => x == OperationType.Replace,
+      new()
+      {
+        {
+          x => string.IsNullOrEmpty(x.value?.ToString()) || DateTime.TryParse(x.value.ToString(), out _),
```

Comment:
Have you checked this? What about "   " or "   2022-10-10 00:00:00   "?
may be 'string.IsNullOrEmpty(x.value?.ToString())' change to 'string.IsNullOrWhiteSpace(x.value?.ToString())'
and 'DateTime.TryParse(x.value.ToString(), out _)' change to 'DateTime.TryParse(x.value.ToString().Trim(), out _)'?
type is 'DateTime?', pipeline should pass only Date or null (and existing code is enough), but, please, check it

Reply:
great! thanks)